app/controllers/api/v1/site_settings_controller.rb

```diff
@@ -3,10 +3,27 @@
 module Api
   module V1
     class SiteSettingsController < ApiController
+      def index
+        site_setting_data = {}
+        site_settings = SiteSetting.joins(:setting).select(:id, :name, :value)
```

Comment:
Great idea on returning a `Hash` of `:name => { :setting_data }` that simplifies exponentially the work on the client app 👏
The use of `:id` however isn't mandatory so I'd vote for the use of settings `:name` to identify records instead since it will remove the need for having to manually iterate over the `site_settings` collection to form the `site_settings_data` `Hash` without introducing a lot of complexity to other APIs and it will also provide a much more informative access logs for deployments (having an access log for `PUT /.../site_settings/Terms.json` is much informative then `/.../site_settings/100.json` ).

The `site_setting_data` can be of one level deep only `Hash` `:name => :value` and formed by something like:
```
Setting.joins(:site_settings)
                               .where(site_settings: { provider: 'greenlight' })
                               .pluck(:name, :value)
                               .to_h
``` 
Also, let's avoid to join on child/join tables unless it's mandatory or helps to simplify our workload in a certain aspect.

Reply:
I see where you are coming from, the reason I used :id though is to cover the case where the provider is not greenlight for example. In that case, you would need to use the name, value, and provider to locate the correct site setting to update. The way I have it now, you only need the :id to update the correct site setting.

That being said, I'm not sure what we want to go with for now. @farhatahmad